### PR TITLE
Add MAX32665HSP board

### PR DESCRIPTION
**Pull request by ozersa**
Signed-off-by: Sadik.Ozer <Sadik.Ozer@maximintegrated.com>

### CODE REVIEW

**Review by 0xc0170 — APPROVED** [review]

File: test/info.py [hunk 1]
```diff
@@ -189,6 +190,7 @@ def VENDOR_TO_FAMILY(x, y) : return (VENDOR_ID[x] <<8) | y
     (   0x0350,     VENDOR_TO_FAMILY('Stub', 1),        'k20dx_xdot_l151_if',                       'k20dx_bl',         None                                    ), # TODO - set target to 'MTS-xDot-L151CC' when mbed-os supports this
     (   0x0360,     VENDOR_TO_FAMILY('NXP', 1),         'lpc4322_hani_iot_if',                      'lpc4322_bl',       'HANI-IOT'                              ),
     (   0x0409,     VENDOR_TO_FAMILY('Stub', 1),        'max32625_max32630fthr_if',                 'max32625_bl',      'MAX32630'                              ),
+    (   0x0409,     VENDOR_TO_FAMILY('Stub', 1),        'max32625_max32665hsp_if',                  'max32625_bl',      'MAX32665'                              ),
```
Comment (jessexm) [inline test/info.py:193]: Board ID likely 0x0419 not 0x409

**Comment by flit** [comment]
@ozersa Can you try using `./tools/post_build_script.py` to generate a custom binary for your new board? (The script currently only works with Python 2.) You just need the .FLM flash algo file from a CMSIS-Pack.

The eventual goal is to not require code to be added to produce firmware binaries for different boards or flash algorithms. The only case where code would need to be modified for new boards is to support a new target family that requires custom connect/reset logic, or to support a new HIC.

**Comment by mathias-arm** [comment]
@ozersa: can you rebase, fix the typo and possibly (as asked in #799) add Maxim copyrights.

In `experimental_compilers` branch the `./tools/post_build_script.py` works with Python3.

**Comment by ozersa** [comment]
> @ozersa: can you rebase, fix the typo and possibly (as asked in #799) add Maxim copyrights.
> 
> In `experimental_compilers` branch the `./tools/post_build_script.py` works with Python3.

Commit rebased with final master point, typo fixed, added Maxim Integrated open source copyright.
 ./tools/post_build_script.py works, run with python 3.7.6 version

**Review by mathias-arm — APPROVED** [review]

**Comment by mbrossard** [comment]
It seems `0x419` was [allocated](https://github.com/ARMmbed/mbed-os-tools/commit/b4c56b10fdbd159e1bd81d06905638101debcf21) to `MAX35103EVKIT2` 3 years ago.

**Comment by ozersa** [comment]
Hello,
We stop to support the "MAX35103EVKIT2" board so that it has been removed from MBED targets after MBED 6.x versions,
Is there any risk if we use the 0x419 number for the new board (MAX32665HSP)?

Regards
Sadik

From: Mathias Brossard ***@***.***>
Sent: Monday, June 14, 2021 7:37 AM
To: ARMmbed/DAPLink ***@***.***>
Cc: Sadik Ozer ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [ARMmbed/DAPLink] Add MAX32665HSP board (#731)

EXTERNAL EMAIL



It seems 0x419 was allocated<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FARMmbed%2Fmbed-os-tools%2Fcommit%2Fb4c56b10fdbd159e1bd81d06905638101debcf21&data=04%7C01%7Csadik.ozer%40maximintegrated.com%7C1f89dc6b55804deb0c5308d92eee1515%7Cfbd909dfea694788a554f24b7854ad03%7C0%7C0%7C637592422357704443%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=SUiQpraWGh%2Be4EOTTHTdk%2FM3zc%2BcLdihZbD%2BjMr7ijE%3D&reserved=0> to MAX35103EVKIT2 3 years ago.

-

**Comment by mathias-arm** [comment]
The problem is that Mbed tooling will probably continue to recognize 0x419 as `MAX35103EVKIT2`.
A new wiki page was added with some information about [Board-IDs](https://github.com/ARMmbed/DAPLink/wiki/Board-IDs). It includes the [link](https://os.mbed.com/request-board-id) to request a Board ID.

**Pull request merged**